contrib/opencensus-ext-stackdriver/opencensus/ext/stackdriver/stats_exporter/__init__.py

```diff
@@ -131,209 +130,174 @@ def options(self):
-            timestamp_end = (end - EPOCH_DATETIME).total_seconds()
-
-            point.interval.end_time.seconds = int(timestamp_end)
-
-            secs = point.interval.end_time.seconds
-            point.interval.end_time.nanos = int((timestamp_end - secs) * 10**9)
-
-            if aggregation_type is not aggregation.Type.LASTVALUE:
-                if timestamp_start == timestamp_end:
-                    # avoiding start_time and end_time to be equal
-                    timestamp_start = timestamp_start - 1
-
-            start_time = point.interval.start_time
-            start_time.seconds = int(timestamp_start)
-            start_secs = start_time.seconds
-            start_time.nanos = int((timestamp_start - start_secs) * 1e9)
-
-            time_series_list.append(series)
-
-        return time_series_list
-
-    def create_metric_descriptor(self, view):
-        """ it creates a MetricDescriptor
-        for the given view data in Stackdriver Monitoring.
-        An error will be raised if there is
-        already a metric descriptor created with the same name
-        but it has a different aggregation or keys.
-        """
-        view_measure = view.measure
-        view_aggregation = view.aggregation
-        view_name = view.name
-
-        metric_type = namespaced_view_name(view_name,
-                                           self.options.metric_prefix)
-        value_type = None
-        unit = view_measure.unit
-        metric_desc = monitoring_v3.enums.MetricDescriptor
-        agg_type = aggregation.Type
-
-        # Default metric Kind
-        metric_kind = metric_desc.MetricKind.CUMULATIVE
-
-        if view_aggregation.aggregation_type is agg_type.COUNT:
-            value_type = metric_desc.ValueType.INT64
-            # If the aggregation type is count
-            # which counts the number of recorded measurements
-            # the unit must be "1", because this view
-            # does not apply to the recorded values.
-            unit = str(1)
-        elif view_aggregation.aggregation_type is agg_type.SUM:
-            if isinstance(view_measure, measure.MeasureInt):
-                value_type = metric_desc.ValueType.INT64
-            if isinstance(view_measure, measure.MeasureFloat):
-                value_type = metric_desc.ValueType.DOUBLE
-        elif view_aggregation.aggregation_type is agg_type.DISTRIBUTION:
-            value_type = metric_desc.ValueType.DISTRIBUTION
-        elif view_aggregation.aggregation_type is agg_type.LASTVALUE:
-            metric_kind = metric_desc.MetricKind.GAUGE
-            if isinstance(view_measure, measure.MeasureInt):
-                value_type = metric_desc.ValueType.INT64
-            if isinstance(view_measure, measure.MeasureFloat):
-                value_type = metric_desc.ValueType.DOUBLE
+    def create_time_series_list(self, metric):
+        tsl = []
+        for ts in metric.time_series:
+            tsl.append(self._convert_series(metric, ts))
```

Comment:
Curious, would this `[self._convert_series(metric, ts) for ts in metric.time_series]` give better readability and perf? And would tuple work here?

Reply:
Changed in bce0b86. No strong opinion on lists vs tuples, but it's fine for the caller to modify this one and we generally use lists internally like this.

Reply:
Got it. Thanks!